src/published-data/published-data.controller.ts

```diff
@@ -483,80 +654,116 @@ export class PublishedDataController {
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+    }
 
     const OAIServerUri = this.configService.get<string>("oaiProviderRoute");
 
     let returnValue = null;
     if (OAIServerUri) {
       returnValue = await this.publishedDataService.resyncOAIPublication(
         id,
-        publishedData,
+        { ...publishedData, ...data },
         OAIServerUri,
       );
     }
 
-    try {
-      await this.publishedDataService.update({ doi: id }, publishedData);
-    } catch (error: any) {
-      throw new HttpException(
-        `Error occurred: ${error}`,
-        error.response?.status || HttpStatus.FAILED_DEPENDENCY,
-      );
-    }
+    await this.publishedDataService.update({ doi: id }, data);
 
     return returnValue;
   }
 }
 
-function formRegistrationXML(publishedData: PublishedData): string {
+function doiRegistrationJSON(publishedData: PublishedData): object {
+  const { title, abstract, metadata, doi } = publishedData;
   const {
-    affiliation,
+    creators,
+    contributors,
+    resourceType,
     publisher,
     publicationYear,
-    title,
-    abstract,
-    resourceType,
-    creator,
-  } = publishedData;
-  const doi = publishedData.doi;
-  const uniqueCreator = creator.filter(
-    (author, i) => creator.indexOf(author) === i,
-  );
-
-  const creatorElements = uniqueCreator.map((author) => {
-    const names = author.split(" ");
-    const firstName = names[0];
-    const lastName = names.slice(1).join(" ");
-
-    return `
-            <creator>
-                <creatorName>${lastName}, ${firstName}</creatorName>
-                <givenName>${firstName}</givenName>
-                <familyName>${lastName}</familyName>
-                <affiliation>${affiliation}</affiliation>
-            </creator>
-        `;
-  });
-
-  return `<?xml version="1.0" encoding="UTF-8"?>
-        <resource xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://datacite.org/schema/kernel-4" xsi:schemaLocation="http://datacite.org/schema/kernel-4 https://schema.datacite.org/meta/kernel-4.4/metadata.xsd">
-            <identifier identifierType="doi">${doi}</identifier>
-            <creators>
-                ${creatorElements.join("\n")}
-            </creators>
-            <titles>
-                <title>${title}</title>
-            </titles>
-            <publisher>${publisher}</publisher>
-            <publicationYear>${publicationYear}</publicationYear>
-            <descriptions>
-                <description xml:lang="en-us" descriptionType="Abstract">${abstract}</description>
-            </descriptions>
-            <resourceType resourceTypeGeneral="Dataset">${resourceType}</resourceType>
-        </resource>
-    `;
+    subjects,
+    descriptions,
+    relatedItems,
+    relatedIdentifiers,
+    language,
+    dates,
+    sizes,
+    formats,
+    geoLocations,
+    fundingReferences,
+    landingPage,
```

Comment:
Given that the metadata is arbitrary, in my opinion is too risky to assume that the fields that are needed to register a DOi are all there.